### PR TITLE
[#2962] feat(catalog-lakehouse-iceberg): Add reserved properties to Iceberg Table Properties in Gravitino

### PR DESCRIPTION
**Pull request by caican00**
### What changes were proposed in this pull request?
Add reserved properties to Table properties when load an Iceberg table, such as:

```
provider,
format,
current-snapshot-id,
location,
format-version,
sort-order,
identifier-fields
```

### Why are the changes needed?
for example, when execute `desc extended IcebergTableName`，it will get some information from the result properties of this Iceberg table, so it should contain the above properties.

Fix: https://github.com/datastrato/gravitino/issues/2962

### Does this PR introduce any user-facing change?
No.

### How was this patch tested?
New UTs and ITs.

### CODE REVIEW

**Comment by caican00** [comment]
Hi @FANNG1 could you please help review this PR when you are free? Thanks!

**Comment by FANNG1** [comment]
> Hi @FANNG1 could you please help review this PR when you are free? Thanks!

ok

File: catalogs/catalog-lakehouse-iceberg/src/main/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/converter/DescribeIcebergSortOrderVisitor.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
+/*
```
Comment (FANNG1) [inline catalogs/catalog-lakehouse-iceberg/src/main/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/converter/DescribeIcebergSortOrderVisitor.java:1]: please use apache header

File: catalogs/catalog-lakehouse-iceberg/src/main/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/converter/DescribeIcebergSortOrderVisitor.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+import org.apache.iceberg.NullOrder;
+import org.apache.iceberg.transforms.SortOrderVisitor;
+
+/**
+ * Convert expressions of Iceberg SortOrders to function string.
+ *
+ * <p>Referred from org/apache/iceberg/spark/Spark3Util/DescribeSortOrderVisitor.java
+ */
+public class DescribeIcebergSortOrderVisitor implements SortOrderVisitor<String> {
```
Comment (FANNG1) [inline catalogs/catalog-lakehouse-iceberg/src/main/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/converter/DescribeIcebergSortOrderVisitor.java:15]: add class to `liCENSE` file

File: catalogs/catalog-lakehouse-iceberg/src/main/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/IcebergTable.java [hunk 3]
```diff
@@ -222,4 +233,39 @@ protected IcebergTable internalBuild() {
   public static Builder builder() {
     return new Builder();
   }
+
+  private static Map<String, String> buildReservedProperties(TableMetadata table) {
```
Comment (FANNG1) [inline catalogs/catalog-lakehouse-iceberg/src/main/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/IcebergTable.java:237]: could you move this logic to a helper or util class with apache header and add to LICENCE?

Comment (caican00) [reply catalogs/catalog-lakehouse-iceberg/src/main/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/IcebergTable.java:237]: @FANNG1 only move this method?

Comment (caican00) [reply catalogs/catalog-lakehouse-iceberg/src/main/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/IcebergTable.java]: fixed.

File: catalogs/catalog-lakehouse-iceberg/src/main/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/converter/DescribeIcebergSortOrderVisitor.java (hunk 1 above)
Comment (caican00) [reply catalogs/catalog-lakehouse-iceberg/src/main/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/converter/DescribeIcebergSortOrderVisitor.java:1]: updated.

File: catalogs/catalog-lakehouse-iceberg/src/main/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/utils/IcebergTablePropertiesUtil.java [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+import com.datastrato.gravitino.catalog.lakehouse.iceberg.converter.DescribeIcebergSortOrderVisitor;
+import com.google.common.base.Joiner;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import org.apache.iceberg.TableMetadata;
+import org.apache.iceberg.TableProperties;
+import org.apache.iceberg.transforms.SortOrderVisitor;
+
```
Comment (FANNG1) [inline catalogs/catalog-lakehouse-iceberg/src/main/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/utils/IcebergTablePropertiesUtil.java:39]: please add Referred xxx

File: catalogs/catalog-lakehouse-iceberg/src/main/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/utils/IcebergTablePropertiesUtil.java [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+public class IcebergTablePropertiesUtil {
```
Comment (FANNG1) [inline catalogs/catalog-lakehouse-iceberg/src/main/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/utils/IcebergTablePropertiesUtil.java:40]: please add private constructor

**Comment by FANNG1** [comment]
LGTM, except 2 comments, @caican00 could you fix it?

File: catalogs/catalog-lakehouse-iceberg/src/main/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/utils/IcebergTablePropertiesUtil.java (hunk 4 above)
Comment (caican00) [reply catalogs/catalog-lakehouse-iceberg/src/main/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/utils/IcebergTablePropertiesUtil.java:39]: updated.

File: catalogs/catalog-lakehouse-iceberg/src/main/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/utils/IcebergTablePropertiesUtil.java (hunk 5 above)
Comment (caican00) [reply catalogs/catalog-lakehouse-iceberg/src/main/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/utils/IcebergTablePropertiesUtil.java:41]: updated.

**Comment by caican00** [comment]
> LGTM, except 2 comments, @caican00 could you fix it?

@FANNG1 all comments have been addressed.

**Review by FANNG1 — APPROVED** [review]

**Comment by FANNG1** [comment]
@caican00 merged to main, thanks for you contribution!

**Pull request merged**